envoyauth/response.go

```diff
@@ -184,19 +185,72 @@ func (result *EvalResult) GetRequestHTTPHeadersToRemove() ([]string, error) {
 	return result.getStringSliceFromDecision("request_headers_to_remove")
 }
 
+func (result *EvalResult) makeHeadersFromDecision(fieldName string, hb headersBaggage) error {
+	switch decision := result.Decision.(type) {
+	case bool:
+		return nil
+	case map[string]interface{}:
+		val, ok := decision[fieldName]
+		if !ok {
+			return nil
+		}
+
+		return transformAnyInputToHeadersBaggage(val, hb)
+	default:
+		return result.invalidDecisionErr()
+	}
+}
+
+type headersBaggage interface {
```

Comment:
```suggestion
type headersContainer interface {
```

I suggest that we name this something other than `headersBaggage`. `headersContainer`, `headersBag`, etc all are fine - baggage (in native english) implies that this is unimportant/unwanted.